src/Symfony/Component/Intl/Data/Provider/TimezoneDataProvider.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Intl\Data\Provider;
+
+use Symfony\Component\Intl\Locale;
+use Symfony\Component\Intl\Data\Bundle\Reader\BundleEntryReaderInterface;
+
+/**
+ * Data provider for timezone-related data.
+ *
+ * @internal
+ */
+class TimezoneDataProvider
+{
+    /**
+     * @var string
+     */
+    private $path;
+
+    /**
+     * @var BundleEntryReaderInterface
+     */
+    private $reader;
+
+    /**
+     * Creates a data provider that reads timezone-related data from a
+     * resource bundle.
+     *
+     * @param string                     $path   The path to the resource bundle.
+     * @param BundleEntryReaderInterface $reader The reader for reading the resource bundle.
+     */
+    public function __construct($path, BundleEntryReaderInterface $reader)
+    {
+        $this->path = $path;
+        $this->reader = $reader;
+    }
+
+    public function getIDs()
+    {
+        return $this->reader->readEntry($this->path, 'meta', array('Timezones'));
+    }
+
+    public function getName($zoneID, $displayLocale = null)
+    {
+        if (null === $displayLocale) {
+            $displayLocale = Locale::getDefault();
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Avoid using static access to class '\Symfony\Component\Intl\Locale' in method 'getName'.](https://www.codacy.com/app/git_9/symfony/file/2332020092/issues/source?bid=3207821&fileBranchId=3207896#l55)